File: ControleDeGastos.API/ControleDeGastos.API.Dominio/Models/Transacao.cs
using ControleDeGastos.API.Dominio.Enumeradores;
using Infraestrutura.ConfiguracaoEntidades;

namespace ControleDeGastos.API.Dominio.Models
{
    /// <summary>
    /// Entidade Categoria, herda de BaseEntity para ser configurada posteriormente.
    /// </summary>
    public class Transacao : BaseEntity
    {
        public Guid CategoriaId { get; set; }
        public Guid PessoaId { get; set; }
        public string Descricao { get; set; }
        public double Valor { get; set; }
        public EnumFinalidade Tipo { get; set; }
        public virtual Categoria Categoria { get; set; }
        public virtual Pessoa Pessoa { get; set; }

        public Transacao() { }

        public Transacao(Guid categoriaId, Guid pessoaId, double valor, int tipo)
        {
            this.CategoriaId = categoriaId;
            this.PessoaId = pessoaId;
            this.Valor = valor;
            this.Tipo = (EnumFinalidade)tipo;
        }
    }
}

File: controle_gastos/src/models/Transacao/Transacao.ts
import type { Categoria } from "../Categoria/Categoria";
import type { Pessoa } from "../Pessoas/Pessoa";
/// <summary>
/// Interface de Transacao
/// </summary>
export interface Transacao {
  id: string;
  categoriaId: string;
  pessoaId: string;
  descricao: string;
  valor: number;
  tipo: number;
  categoria: Categoria;
  pessoa: Pessoa
}
